books/kestrel/c/syntax/validator.lisp

```diff
@@ -219,6 +223,58 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+(define valid-lookup-externals ((ident identp) (table valid-tablep))
+  :returns (info? valid-external-info-optionp)
+  :short "Look up the validation information of an identifier in the
+          @('externals') map."
+  :long
+  (xdoc::topstring
+   (xdoc::p
+    "This holds the validation information
+     for an identifier with external linkage
+     which has been declared in any scope or translation unit.
+     See @(see valid-table)."))
+  (b* (((valid-table table) table))
+    (cdr (omap::assoc (ident-fix ident) table.externals)))
+  :hooks (:fix))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(define valid-update-externals ((ident identp)
```

Comment:
Here the `update` is appropriate, as opposed to `add` in `valid-add-ord`, which _adds_ an identifier. Here, instead, we may add it, or update the one that's already there.

You could also consider `valid-update-ext`, if you decide to take the other name shortening suggestions.